moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -538,6 +538,10 @@ bool Trajectory::integrateForward(std::list<TrajectoryStep>& trajectory, double
 
     if (next_discontinuity != switching_points.end() && path_pos > next_discontinuity->first)
     {
+      if (path_pos - next_discontinuity->first < 1e-8)
```

Comment:
Please add comments explaining why this is here and why it fixes the problem. I'm having a really hard time following what this method is doing (Which makes it all the more impressive that you found this bug!) and this adds to the complexity.

While you are at it, any additional comments you can add to make this feature more maintainable would be greatly appreciated.

Reply:
Also, seeing ass `time_step_` is hardcoded to `0.001`, I think that changing this to `time_step_ * 1e-5` would at least scale with `time_step_`.